### PR TITLE
tc01_User can log into Railway with valid username and password

### PR DESCRIPTION
**Pull request by taile47711**

### CODE REVIEW

File: src/test/java/tests/BaseTest.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+import common.Constant;
+import helper.DriverHelper;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+
+public class BaseTest {
+    @BeforeMethod
+    public void setup(){
+        DriverHelper.initDrive();
```
Comment (tanmle) [inline src/test/java/tests/BaseTest.java]: spelling error, `Driver` not `Drive`

File: src/test/java/tests/LoginTest.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+import common.Constant;
+import helper.LogHelper;
+import org.slf4j.Logger;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+import page_object.LoginPage;
+
+public class LoginTest extends BaseTest {
+    LoginPage loginPage = new LoginPage();
+    String errorMessage = "There was a problem with your login and/or errors exist in your form.";
```
Comment (tanmle) [inline src/test/java/tests/LoginTest.java]: why do we need this variable?

File: src/test/java/tests/LoginTest.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    public void tc001_UserCanLogIntoRailwayWithValidUsernameAndPassword() {
+        LogHelper.info("ckick tab login");
+        loginPage.clickLoginTab();
+        LogHelper.info("Enter valid data and click Login button");
+        loginPage.login(Constant.USERNAME, Constant.PASSWORD);
+        String actual = loginPage.getWellcomeMessage();
+
+        Assert.assertEquals(actual, "Welcome to Safe Railway", "Message is not displayed as expected");
+    }
+
```
Comment (tanmle) [inline src/test/java/tests/LoginTest.java:22]: unneeded downline

File: src/test/java/tests/LoginTest.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+import org.testng.annotations.Test;
+import page_object.LoginPage;
+
+public class LoginTest extends BaseTest {
+    LoginPage loginPage = new LoginPage();
+    String errorMessage = "There was a problem with your login and/or errors exist in your form.";
+
+    @Test(description = "User can log into Railway with valid username and password")
+    public void tc001_UserCanLogIntoRailwayWithValidUsernameAndPassword() {
+        LogHelper.info("ckick tab login");
```
Comment (tanmle) [inline src/test/java/tests/LoginTest.java]: spelling error.

File: src/test/java/tests/LoginTest.java [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+        loginPage.clickLoginTab();
+        LogHelper.info("Enter valid data and click Login button");
+        loginPage.login(Constant.USERNAME, Constant.PASSWORD);
+        String actual = loginPage.getWellcomeMessage();
+
+        Assert.assertEquals(actual, "Welcome to Safe Railway", "Message is not displayed as expected");
```
Comment (tanmle) [inline src/test/java/tests/LoginTest.java]: this message should be more detail

File: src/test/java/tests/LoginTest.java (hunk 2 above)
Comment (taile47711) [reply src/test/java/tests/LoginTest.java]: fixed

File: src/test/java/tests/LoginTest.java (hunk 5 above)
Comment (MinhQuan119) [inline src/test/java/tests/LoginTest.java]: Should add log step for this Verify Point

**Review by MinhQuan119 — CHANGES_REQUESTED** [review]

**Review by MinhQuan119 — APPROVED** [review]

File: src/test/java/tests/LoginTest.java (hunk 5 above)
Comment (taile47711) [reply src/test/java/tests/LoginTest.java]: fixed

File: src/test/java/tests/BaseTest.java (hunk 1 above)
Comment (taile47711) [reply src/test/java/tests/BaseTest.java]: fixed

File: src/test/java/tests/LoginTest.java (hunk 3 above)
Comment (taile47711) [reply src/test/java/tests/LoginTest.java:22]: fixed

File: src/test/java/tests/LoginTest.java (hunk 4 above)
Comment (taile47711) [reply src/test/java/tests/LoginTest.java]: fixed

File: src/test/java/tests/LoginTest.java (hunk 5 above)
Comment (taile47711) [reply src/test/java/tests/LoginTest.java]: fixed

**Review by tanmle — APPROVED** [review]

**Pull request merged**